File: 100 days of leetcode/Day16.cpp
class Solution {
public:
    string reverseWords(string s) {
        int x = s.length();

        string ans = "";

        int k = x - 1;
        stringstream iss(s);
        string word;

        iss >> word;
        ans += word;

        while(iss >> word){
            ans = word + " " + ans;
        }

        // for (int i = x - 1; i >= 0; i--){
        //     while(s[i] != ' '){
        //         k--;
        //     }
        //     ans = ans + " " + s.substr(k, i);
        //     i -= k;
        // }

        return ans;
    }
};

File: 100 days of leetcode/Day16.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        li = s.split()
        li.reverse()
        ans = " ".join(li)
        return ans


class Solution:
    def reverseWords(self, s: str) -> str:
        return ' '.join(s.split()[::-1])